packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.mm

```diff
@@ -208,6 +208,16 @@ - (void)host:(RCTHost *)host
   }
 }
 
+- (void)loadSourceForHost:(RCTHost *)host onProgress:(RCTSourceLoadProgressBlock)onProgress onComplete:(RCTSourceLoadBlock)loadCallback {
+  if (self->_configuration.loadSourceWithProgressForHost) {
+    self->_configuration.loadSourceWithProgressForHost(host, onProgress, loadCallback);
+  } else if (self->_configuration.loadSourceForHost) {
+    self->_configuration.loadSourceForHost(host, loadCallback);
+  } else {
+    [host loadBundle:onProgress onComplete:loadCallback];
+  }
+}
+
```

Comment:
Hey Riccardo! Feedback makes sense but I think the RootViewFactory makes itself the hostDelegate and not the RCTAppDelegate so it would need to implement this new function?